### PR TITLE
initial analytics dashboard layout and routes

### PR DESCRIPTION
**Pull request by danielogen**
1. What does this PR do?
Shows initial dashboard layout, routes and folder structure (features-based grouping)
2. What JIRA task is related to this PR?
https://airqoteam.atlassian.net/browse/NLYTCS-3
3. How do I test out this PR?
Git clone this repo `cd AirQo-frontend/analytics`, then run `npm install` to get all the necessary dependencies. Afterwards run `npm start` to test out the development server
4. Which end points should are ready for testing?
There are no end points for this PR

### CODE REVIEW

**Comment by danielogen** [comment]
@lilly-muyama: 

> Just a question: shouldn't there be a view for Map as well?

Thanks for the comment. Map view is available as a dashboard component ( go to ... views/Dashboard/components/)

**Comment by lilly-muyama** [comment]
Hi Daniel,

I actually saw it. I just forgot to delete the comment before approving.
Thanks for the good work.

**Comment by danielogen** [comment]
> Looks good to me, except some few files that I think should be included in .gitignore (.editorconfig, .eslintrc)

@sserurich: Thanks I will add them ASAP

File: analytics/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
```
Comment (Baalmart) [inline analytics/.eslintrc]: Great to see the usage of a linting utility, awesome! We shall make great usage of it when enforcing our coding guidelines that we shall agree on as a team.

File: analytics/src/views/Dashboard/components/Map/Map.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
```
Comment (Baalmart) [inline analytics/src/views/Dashboard/components/Map/Map.js]: If we are taking the approach of "features first" under the "views" directory, might we consider "Map" to be a feature of its own?

File: analytics/src/layouts/Main/Main.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+const Main = props => {
+  const { children } = props;
+
+  const classes = useStyles();
+  const theme = useTheme();
+  const isDesktop = useMediaQuery(theme.breakpoints.up('lg'), {
+    defaultMatches: true
+  });
+
+  const [openSidebar, setOpenSidebar] = useState(false);
```
Comment (Baalmart) [inline analytics/src/layouts/Main/Main.js]: So we are dealing with state at component level. Could we be able to also utilise Redux in the same project? Do you foresee any need for storing application state using Redux? Any concerns or learnings?

File: analytics/src/components/SearchInput/SearchInput.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (Baalmart) [inline analytics/src/components/SearchInput/SearchInput.js]: How shall we be testing these awesome components? Shall we be adding files like SearchInput.test.js under the same directory? We could explore the usage of Jest and Enzyme for unit testing within our awesome project - I have come to understand that they complement one another when it comes to testing. :)

File: analytics/src/layouts/Main/Main.js (hunk 3 above)
Comment (Baalmart) [reply analytics/src/layouts/Main/Main.js]: A medium blog discussing about React Hooks and Redux: https://medium.com/javascript-scene/do-react-hooks-replace-redux-210bab340672

File: analytics/src/components/SearchInput/SearchInput.js (hunk 4 above)
Comment (Baalmart) [reply analytics/src/components/SearchInput/SearchInput.js]: Testing React with Jest and Enzyme: https://medium.com/codeclan/testing-react-with-jest-and-enzyme-20505fec4675

**Pull request merged**